src/sage/combinat/composition.py

```diff
@@ -1754,7 +1756,7 @@ def _element_constructor_(self, lst) -> Composition:
             sage: P([3,3,1]) # indirect doctest
             [3, 3, 1]
         """
-        if isinstance(lst, Composition):
+        if isinstance(lst, (Composition, Partition)):
```

Comment:
a doctest for that, maybe ?

Reply:
I thought I added one here, but I forgot. (It was tested indirectly, but direct tests are good.)